cognite/client/data_classes/data_modeling/instances.py

```diff
@@ -1192,3 +1263,133 @@ def _load(cls, resource: dict[str, Any], cognite_client: CogniteClient | None =
             if "externalId" in resource["unit"]
             else UnitSystemReference.load(resource["unit"]),
         )
+
+
+@dataclass
+class TypePropertyDefinition(CogniteObject):
+    type: PropertyType
+    nullable: bool = True
+    auto_increment: bool = False
+    immutable: bool = False
+    default_value: str | int | dict | None = None
+    name: str | None = None
+    description: str | None = None
+
```

Comment:
This class is almost identical to `ContainerProperty` and `MappedProperty` in view.

Why create a new class?

Since `MappedProperty` have container_property_identifier and container it doesn't fit with this. In addition, that is part of a class hierarchy `ViewProperty` which is very useful for example in `pygen`.

It also feels wrong to combine this with `ContainerProperty` but Im am not certain.

Reply:
No objections 😉